synapse/lib/storm.py

```diff
@@ -4592,6 +4595,22 @@ async def execStormCmd(self, runt, genr):
 
             runt.layerConfirm(('node', 'del', node.form.name))
 
+            if deledges:
+                async with await s_spooled.Set.anit(dirn=self.runt.snap.core.dirn) as edges:
+                    seenverbs = set()
+
+                    async for (verb, n2iden) in node.iterEdgesN2():
+                        if verb not in seenverbs:
+                            runt.layerConfirm(('node', 'edge', 'del', verb))
+                            seenverbs.add(verb)
+                        await edges.add((verb, n2iden))
+
+                    async with self.runt.snap.getEditor() as editor:
+                        async for (verb, n2iden) in edges:
+                            n2 = await editor.getNodeByBuid(s_common.uhex(n2iden))
+                            if n2 is not None:
+                                await n2.delEdge(verb, node.iden())
```

Comment:
To prevent the editor from aggregating an excessive amount of edits we should probably check the number of protonodes and save/clear them if necessary, something like:
```
if await n2.delEdge(verb, node.iden()) and len(editor.protonodes) >= 1000:
    await snap.applyNodeEdits(editor.getNodeEdits())
    editor.protonodes.clear()
```

Reply:
I debated this after your initial comment but thought it wouldn't be needed. Thanks for the snippet :)